Makefile

```diff
@@ -4,7 +4,16 @@ GENERATOR_FOLDER=Generator
 GENERATOR_ARCHIVE_PATH=$(shell cd $(GENERATOR_FOLDER) && swift build $(SWIFT_BUILD_FLAGS) --show-bin-path)/needle
 GENERATOR_VERSION_FOLDER_PATH=$(GENERATOR_FOLDER)/Sources/needle
 GENERATOR_VERSION_FILE_PATH=$(GENERATOR_VERSION_FOLDER_PATH)/Version.swift
-SWIFT_BUILD_FLAGS=--disable-sandbox -c release -Xswiftc -static-stdlib
+SWIFT_BUILD_FLAGS=--disable-sandbox -c release
+
+# Taken from https://github.com/jpsim/SourceKitten/blob/d90296e473d17d1fecbebacfa9b0476682a02456/Makefile
+UNAME=$(shell uname)
+ifeq ($(UNAME), Darwin)
+USE_SWIFT_STATIC_STDLIB:=$(shell test -d $$(dirname $$(xcrun --find swift))/../lib/swift_static/macosx && echo yes)
```

Comment:
seems strange to tie the build flag to the existence of a local directory, but if it's working for out people, I guess this is ok for now.